terraform/modules/s3/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "juju_application" "minio" {
+  # Coordinator requires s3
+  name  = var.minio_app
+  model = var.model_name
+  trust = true
+
+  charm {
+    name    = var.minio_app
+    channel = var.channel
+  }
+  units = 1
+
+  config = {
+    access-key = var.minio_user
+    secret-key = var.minio_password
+  }
```

Comment:
You can also change storage:
storage_directives
https://registry.terraform.io/providers/juju/juju/latest/docs/resources/application

Reply:
Oh, this is indeed interesting...

I need to better understand the best way to deploy MinIO as part of COS.

@sed-i any thoughts bout this?

Reply:
Not sure what it does / implications, but the s3 storage is external to juju, i.e. it is not controlled by juju metadata etc.
Is that what you meant @amandahla?

Reply:
Captured in a new issue: https://github.com/canonical/observability/issues/236